03 Writing Algorithms/01 Key Concepts/04 Multi-Asset Modeling/99 Examples.html

```diff
@@ -0,0 +1,68 @@
+<p>The following examples demonstrate some common practices for multi-asset modeling.</p>
+
+<h4>Example 1: Futures Trading On BTC Portfolio</h4>
+<p>This algorithm demonstrates trading maximum-allowed order quantity of E-mini SP500 Future with an account holding only BTC cash position. Lean will automatically convert the BTC cash into base currency of the Future contract and calculate the maximum order quantity under the Future's buying power model.</p>
+<div class="section-example-container">
+    <pre class="csharp">public class MultiAssetModelingAlgorithm : QCAlgorithm
+{
+    private Future _future;
+
+    public override void Initialize()
+    {
+        SetStartDate(2021, 2, 1);
+        SetEndDate(2021, 3, 1);
+        // Simulate the portfolio is holding BTC cash initially.
+        // Note that the performance will also be affected by the BTC performance in the default account currency.
+        SetCash("BTC", 1000);
+
+        // Request ES future data for trading.
+        _future = AddFuture("ES", Resolution.Daily,
+            extendedMarketHours: true,
+            dataNormalizationMode: DataNormalizationMode.BackwardsRatio,
+            dataMappingMode: DataMappingMode.OpenInterest,
+            contractDepthOffset: 0);
+    }
+    
+    public override void OnData(Slice slice)
+    {
+        if (!Portfolio.Invested)
+        {
+            // Find the maximum order quantity given the current Future buying power model, using all the buying power we have.
+            var parameter = new GetMaximumOrderQuantityForTargetBuyingPowerParameters(Portfolio, _future, 1m, 0m);
+            var maximumQuantityResult = _future.BuyingPowerModel.GetMaximumOrderQuantityForTargetBuyingPower(parameter);
```

Comment:
Set cash account type - forces you to manually handle conversions of BTC -> USD to purchase the future. 

Get initial BTC (from the cashbook)  
Use initial margin cost of the contract in USD? 
To calculate the order size on ES Futures" ? -> Using initial contract margin requirements. This will be in USD/
Free up USD by selling BTC.
Place order for future. 

Use minute resolution data for all demos.

Goal DO an arb strategy from BTCUSD and BTC Futures